### ISSUE TITLE
Parsing repo info

### ISSUE THREAD

**Issue by spyoungtech** [docker-library/repo-info#41]
Thank you so much for your work on this.

I'm wondering if there's a known tool or method to parse the repo info MD files or if the data is available in another format, like JSON.

I'm wanting to parse this data to build a tool to assist docker users in being able to re-constitute old docker builds. For example, by inspecting an image, being able to determine the tag(s)/SHA256 image digest of the upstream base image(s).

**Comment by spyoungtech** [comment]
I made a short Python script to extract some of the info I was looking for from the MD files.

Can be seen here: https://gist.github.com/spyoungtech/3506d5ae9a9888ec709c8fcad33cfc34

Could be extended to parse additional info, I suppose.

**Comment by tianon** [comment]
Yeah, this is something I've thought a lot about (and I know a few people have done privately in the past).

Ideally, this whole repository would be replaced by a real proper service with an API (and boatloads of historical data), but that's a bit more complicated. :sweat_smile:

Some of this is something Docker Hub really ought to provide out of the box (IMO), but it's not exactly straightforward for them to do so, I think.

**Comment by captn3m0** [comment]
@spyoungtech Did you end up publishing this somewhere?

**Comment by Jircs1** [comment]
👍
